### This file output the txt file to make sure each line is less than 200. 

In [13]:
import os

In [14]:
def analyze_files(folder_path):
    total_tokens = 0
    total_lines = 0
    lines_less_than_50 = 0
    lines_more_than_50 = 0

    for filename in os.listdir(folder_path):
        file_path = os.path.join(folder_path, filename)
        if os.path.isfile(file_path):
            with open(file_path, 'r') as file:
                for line in file:
                    line = line.strip()
                    if line:
                        total_lines += 1
                        tokens = line.split()
                        token_count = len(tokens)
                        total_tokens += token_count
                        if token_count < 50:
                            lines_less_than_50 += 1
                        else:
                            lines_more_than_50 += 1
    ratio_less_than_50 = lines_less_than_50 / total_lines if total_lines > 0 else 0
    print(f"Total tokens: {total_tokens}")
    print(f"Total non-empty lines: {total_lines}")
    print(f"Lines with less than 50 tokens: {lines_less_than_50}")
    print(f"Lines with more than 50 tokens: {lines_more_than_50}")
    print(f"Ratio of lines with less than 50 tokens: {ratio_less_than_50:.2f}")



In [15]:
# Specify the folder path
folder_path = "../original_dataset/gdl_pb_formatted/i386/"

# Call the function to analyze the files
analyze_files(folder_path)

Total tokens: 67815945
Total non-empty lines: 2803557
Lines with less than 50 tokens: 2495867
Lines with more than 50 tokens: 307690
Ratio of lines with less than 50 tokens: 0.89


x86 token summary: 

Total tokens: 59023799

Total non-empty lines: 2789119

Lines with less than 50 tokens: 2557702

Lines with more than 50 tokens: 231417

Ratio of lines with less than 50 tokens: 0.92

Arm32 token summary:

Total tokens: 150894537

Total non-empty lines: 5812795

Lines with less than 50 tokens: 5280755

Lines with more than 50 tokens: 532040

Ratio of lines with less than 50 tokens: 0.91

Arm64 Token summary:

Total tokens: 171768673

Total non-empty lines: 7413083

Lines with less than 50 tokens: 6829128

Lines with more than 50 tokens: 583955

Ratio of lines with less than 50 tokens: 0.92

i386 token summary: 

Total tokens: 67815945

Total non-empty lines: 2803557

Lines with less than 50 tokens: 2495867

Lines with more than 50 tokens: 307690

Ratio of lines with less than 50 tokens: 0.89